frontend/src/App.vue

```diff
@@ -1,20 +1,28 @@
 <script setup lang="ts">
-// Vue
-import { onErrorCaptured } from 'vue';
-import { RouterView } from 'vue-router';
-// PrimeVue
+import { storeToRefs } from 'pinia';
 import ConfirmDialog from 'primevue/confirmdialog';
-import ProgressBar from 'primevue/progressbar';
 import Toast from 'primevue/toast';
 import { useToast } from 'primevue/usetoast';
-// Components
-import AppLayout from '@/components/layout/AppLayout.vue';
-import InitApp from '@/components/layout/InitApp.vue';
-import Navbar from '@/components/layout/Navbar.vue';
+import { onBeforeMount, onErrorCaptured } from 'vue';
+import { RouterView } from 'vue-router';
+
+import { AppLayout, Navbar, ProgressLoader } from '@/components/layout';
+import { useAppStore, useAuthStore, useUserStore } from '@/store';
+
+const appStore = useAppStore();
+const { getIsLoading } = storeToRefs(appStore);
+
+onBeforeMount(async () => {
+  appStore.beginDeterminateLoading();
+  await useAuthStore().init();
```

Comment:
i thought useAuthStore().init(); was already called when you initializeApp() in main.ts

Reply:
We might not necessarily need it here as in theory this should have either already been done in main.ts or done in children components; App.vue itself has no requirements to know about the authentication state; but the children components do.